src/rstar.jl

```diff
@@ -0,0 +1,93 @@
+using .XGBoost
+"""
+    rstar(chains::Chains; subset = 0.8, niter = 1_000, eta = 0.5, XGBoostParams)
+    rstar(chains::Chains, iterations::Int; subset = 0.8, niter = 1_000, eta = 0.5, XGBoostParams)
+    rstar(x::AbstractMatrix, y::AbstractVector, nchains::Int, iterations::Int; subset = 0.8, niter = 1_000, eta = 0.5, XGBoostParams)
+
+Compute the R* statistic for convergence diagnostic of MCMC. This implementation is an adaption of Algorithm 1 & 2, described in [Lambert & Vehtari]. Note that the correctness of the statistic depends on the convergence of the classifier used internally in the statistic. You can track if the training of the classifier converged by inspection of the printed RMSE values from the XGBoost backend. To adjust the number of iterations used to train the classifier set `niter` accordingly.
+
+# Usage
+
+```julia-repl
+using XGBoost
+# You need to load XGBoost before using MCMCChains.Rstar
+
+...
+
+chn = ...
+
+# Compute R⋆ using defaults settings for the  gradient boosting classifier used to compute the statistic.
+# This is the recomended use.
+R = rstar(chn)
+
+# Compute 100 samples of the R⋆ statistic using sampling from according to the prediction probabilities.
+# This approach can be slow and results in a less accurate estimation of the R* statistic.
+# See discussion in Section 3.1.3 in the paper.
+Rs = rstar(chn, 100)
+
+# estimate Rstar
+R = mean(Rs)
+
+# visualize distribution
+histogram(Rs)
+```
+
+## References:
+[Lambert & Vehtari] Ben Lambert and Aki Vehtari. "R∗: A robust MCMC convergence diagnostic with uncertainty using gradient-boostined machines." Arxiv 2020.
+"""
+function rstar(x::AbstractMatrix, y::AbstractVector, nchains::Int, iterations::Int; subset = 0.8, niter = 1_000, eta = 0.5, xgboostparams...)
+
+    N = length(y)
+
+    # randomly sub-select training and testing set
+    Ntrain = round(Int, N*subset)
+    Ntest = N - Ntrain
+
+    ids = Random.shuffle(collect(1:N))
+    train_ids = ids[1:Ntrain]
+    test_ids = ids[(Ntrain+1):end]
```

Comment:
```suggestion
    ids = Random.randperm(N)
    train_ids = view(ids, 1:Ntrain)
    test_ids = view(ids, (Ntrain+1):N)
```
Additionally, it would be nice to make a `rng::AbstractRNG` the first argument to `rstar` and use the provided random number generator here. Additionally, maybe a (additional) function that takes explicit training and test data instead of splitting the provided data might be useful.

Reply:
Nice one, thanks.